front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/index.ts

```diff
@@ -0,0 +1,147 @@
+  res: NextApiResponse<WithAPIErrorResponse<GetDataSourceViewResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const dataSourceView = await DataSourceViewResource.fetchById(
+    auth,
+    req.query.dsvId as string
+  );
+
+  const dataSource = dataSourceView?.dataSource;
+  const vault = dataSourceView?.vault;
+
+  if (
+    !dataSourceView ||
+    !dataSource ||
+    !vault ||
+    req.query.vId !== vault.sId ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "read"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+  switch (req.method) {
+    case "GET": {
+      return res.status(200).json({
+        dataSourceView: getDataSourceViewInfo(dataSourceView),
+      });
+    }
+    case "PATCH": {
+      if (!auth.isAdmin() || !auth.isBuilder()) {
+        // Only admins, or builders who have access to the vault, can patch
+        return apiError(req, res, {
+          status_code: 403,
+          api_error: {
+            type: "workspace_auth_error",
+            message:
+              "Only users that are `admins` or `builder` can administrate vaults.",
+          },
+        });
+      }
+
+      const bodyValidation = PatchDataSourceViewSchema.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${pathError}`,
+          },
+        });
+      }
+
+      const { parentsIn } = bodyValidation.right;
+      const updateResult = await dataSourceView.updateParents(auth, parentsIn);
+
+      if (updateResult.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "The data source view cannot be updated.",
+          },
+        });
+      }
+      return res
+        .status(200)
+        .json({ dataSourceView: getDataSourceViewInfo(dataSourceView) });
+    }
+    case "DELETE": {
+      if (!auth.isAdmin() || !auth.isBuilder()) {
+        // Only admins, or builders who have to the vault, can patch
+        return apiError(req, res, {
+          status_code: 403,
+          api_error: {
+            type: "workspace_auth_error",
+            message:
+              "Only users that are `admins` or `builder` can administrate vaults.",
+          },
+        });
+      }
+
+      const deleteResult = await dataSourceView.delete(auth);
```

Comment:
We should not allow to do this, if it's sill used by some assistants. Also we should ensure that this view does not belong to the global vault.

Reply:
Why can't we delete a view from the global vault ? It will probably be common to have connections that only appears in private vault and that we don't want to see in global

Reply:
🤔 Given that we create those default views for each managed (and connected) data sources. I'm not sure we should allow people to delete them at all.